ambry-clustermap/src/main/java/com/github/ambry/clustermap/ClusterMapUtils.java

```diff
@@ -548,22 +551,26 @@ private boolean areAllLocalReplicasForPartitionUp(PartitionId partitionId) {
       try {
         if (partitionClass == null) {
           toReturn.addAll(allPartitions);
-        } else if (partitionIdsByClassAndLocalReplicaCount.containsKey(partitionClass)) {
+        } else {
           SortedMap<Integer, List<PartitionId>> partitionsByReplicaCount =
               partitionIdsByClassAndLocalReplicaCount.get(partitionClass);
+          if (partitionsByReplicaCount == null) {
+            partitionsByReplicaCount = partitionIdsByClassAndLocalReplicaCount.get(defaultPartitionClass);
+          }
+          if (partitionsByReplicaCount == null) {
+            throw new IllegalArgumentException(
+                "No partitions for default partition class " + partitionClass + " found");
```

Comment:
minor: `"No partitions for partition class = '" + partitionClass + "' and default partition class = '" + defaultPartitionClass + "' found"`